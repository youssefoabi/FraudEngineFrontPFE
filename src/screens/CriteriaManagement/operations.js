import { all, call, put, select, takeLatest } from 'redux-saga/effects';
import {
  contains,
  filter,
  isEmpty,
  isNil,
  join,
  length,
  map,
  path,
  pathOr,
  pipe,
  prop,
  propEq,
  values,
} from 'ramda';

import {
  DELETE_CRITERIA_ERROR,
  DELETE_CRITERIA_REQUEST,
  DELETE_CRITERIA_SUCCESS,
  FETCH_CRITERIA,
  FETCH_CRITERIA_ERROR,
  SET_CRITERIA,
  SET_CRITERIA_SELECTED,
} from '../../components/CriteriaDataGrid/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';

export const parseResult = pathOr([], ['data']);

export function* fetchCriteria() {
  try {
    const result = yield call(clientApi().getCriteria);

    const criteria = parseResult(result);

    yield put({ type: SET_CRITERIA, payload: criteria });

    yield put({ type: SET_CRITERIA_SELECTED, payload: [] });
  } catch (error) {
    yield put({ type: FETCH_CRITERIA_ERROR, error });
  }
}

export function* deleteCriterion(id) {
  try {
    yield call(clientApi().deleteCriterion, id);
  } catch (error) {
    const name = yield select(path(['entities', 'criteria', 'byId', id, 'name']));
    return { isSuccess: false, name, id };
  }

  return { isSuccess: true, id };
}

export function* deleteCriteria() {
  yield put({ type: PENDING_REQUEST });

  try {
    const selected = yield select(path(['navigation', 'criteria', 'selected']));

    if (selected) {
      const deletedCriteriaState = yield all(map(id => call(deleteCriterion, id), selected));
      const errors = pipe(filter(propEq('isSuccess', false)), map(prop('name')))(deletedCriteriaState);

      const hasError = !isNil(errors) && !isEmpty(errors);
      const hasMoreThanOneError = hasError && length(errors) > 1;
      if (hasError)
        yield put({
          type: DELETE_CRITERIA_ERROR,
          error: {
            errorMessage: `Echec de suppression d${hasMoreThanOneError ? 'es' : 'u'}  critÃ¨re${
              hasMoreThanOneError ? 's' : ''
            } ${join(',', errors)}`,
          },
        });

      const success = pipe(filter(propEq('isSuccess', true)), map(prop('id')))(deletedCriteriaState);

      if (isNil(success) || isEmpty(success)) {
        yield put({ type: RESET_PENDING_REQUEST });
        return null;
      }

      const criteria = yield select(pipe(path(['entities', 'criteria', 'byId']), values));
      const newCriteria =
        !isNil(criteria) && criteria.filter(criterion => !contains(prop('id', criterion), success));
      yield put({ type: SET_CRITERIA, payload: newCriteria });

      yield put({ type: SET_CRITERIA_SELECTED, payload: [] });
      if (!hasError) yield put({ type: DELETE_CRITERIA_SUCCESS });
    }
  } catch (error) {
    yield put({
      type: DELETE_CRITERIA_ERROR,
      error: { errorMessage: prop('message', error) },
    });
  }

  yield put({ type: RESET_PENDING_REQUEST });
  return null;
}

export default [
  takeLatest(FETCH_CRITERIA, fetchCriteria),
  takeLatest(DELETE_CRITERIA_REQUEST, deleteCriteria),
];
