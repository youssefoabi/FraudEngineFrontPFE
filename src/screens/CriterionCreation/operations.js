import { call, put, takeLatest } from 'redux-saga/effects';
import { isNilOrEmpty } from 'ramda-adjunct';

import {
  CREATE_CRITERION_ERROR,
  CREATE_CRITERION_REQUEST,
  CREATE_CRITERION_SUCCESS,
} from '../../components/CriterionForm/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';
import parse from './parse';

export function* createCriterion({ payload }) {
  yield put({ type: PENDING_REQUEST });

  try {
    const { criterion } = payload;
    if (isNilOrEmpty(criterion)) return null;

    yield call(clientApi().createCriterion, parse(criterion));

    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: CREATE_CRITERION_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: CREATE_CRITERION_ERROR,
      error: { errorMessage: "Erreur d'enregistrement du crit√®re" },
    });
  }

  return true;
}

export default [takeLatest(CREATE_CRITERION_REQUEST, createCriterion)];
