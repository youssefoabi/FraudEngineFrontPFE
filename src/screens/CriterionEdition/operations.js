import { call, put, takeLatest } from 'redux-saga/effects';

import {
  EDIT_CRITERION_ERROR,
  EDIT_CRITERION_REQUEST,
  EDIT_CRITERION_SUCCESS,
} from '../../components/CriterionForm/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';
import parse from '../CriterionCreation/parse';

export function* saveCriterion({ payload }) {
  yield put({ type: PENDING_REQUEST });

  try {
    const { criterion } = payload;
    if (!criterion) return null;

    yield call(clientApi().updateCriterion, parse(criterion));
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: EDIT_CRITERION_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: EDIT_CRITERION_ERROR,
      error: { errorMessage: "Erreur d'enregistrement de moteur de recherche" },
    });
  }

  return true;
}

export default [takeLatest(EDIT_CRITERION_REQUEST, saveCriterion)];
