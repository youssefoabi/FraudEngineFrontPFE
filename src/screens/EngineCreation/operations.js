import { call, put, takeLatest } from 'redux-saga/effects';
import { isNilOrEmpty } from 'ramda-adjunct';

import {
  CREATE_ENGINE_ERROR,
  CREATE_ENGINE_REQUEST,
  CREATE_ENGINE_SUCCESS,
} from '../../components/EngineForm/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';
import parse from './parse';

export function* createEngine({ payload }) {
  yield put({ type: PENDING_REQUEST });

  try {
    const { engine } = payload;
    if (isNilOrEmpty(engine)) return null;

    yield call(clientApi().createEngine, parse(engine));

    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: CREATE_ENGINE_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: CREATE_ENGINE_ERROR,
      error: { errorMessage: "Erreur d'enregistrement du moteur de recherche" },
    });
  }

  return true;
}

export default [takeLatest(CREATE_ENGINE_REQUEST, createEngine)];
