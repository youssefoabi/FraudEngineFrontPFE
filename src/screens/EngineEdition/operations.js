import { call, put, takeLatest } from 'redux-saga/effects';

import {
  EDIT_ENGINE_ERROR,
  EDIT_ENGINE_REQUEST,
  EDIT_ENGINE_SUCCESS,
} from '../../components/EngineForm/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';
import parse from '../EngineCreation/parse';

export function* saveEngine({ payload }) {
  yield put({ type: PENDING_REQUEST });

  try {
    const { engine } = payload;
    if (!engine) return null;

    yield call(clientApi().updateEngine, parse(engine));
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({ type: EDIT_ENGINE_SUCCESS });
  } catch (error) {
    yield put({ type: RESET_PENDING_REQUEST });
    yield put({
      type: EDIT_ENGINE_ERROR,
      error: { errorMessage: "Erreur d'enregistrement de moteur de recherche" },
    });
  }

  return true;
}

export default [takeLatest(EDIT_ENGINE_REQUEST, saveEngine)];
