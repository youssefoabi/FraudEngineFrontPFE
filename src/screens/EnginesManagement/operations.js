import { all, call, put, select, takeLatest } from 'redux-saga/effects';
import {
  contains,
  filter,
  isEmpty,
  isNil,
  join,
  length,
  map,
  path,
  pathOr,
  pipe,
  prop,
  propEq,
  values,
} from 'ramda';

import {
  DELETE_ENGINES_ERROR,
  DELETE_ENGINES_REQUEST,
  DELETE_ENGINES_SUCCESS,
  FETCH_ENGINES,
  FETCH_ENGINES_ERROR,
  SET_ENGINES,
  SET_ENGINES_SELECTED,
} from '../../components/EnginesDataGrid/actions';
import { PENDING_REQUEST, RESET_PENDING_REQUEST } from '../../components/ServerActionBackdrop/actions';
import clientApi from '../../client-api';

export const parseResult = pathOr([], ['data']);

export function* fetchEngines() {
  try {
    const result = yield call(clientApi().getEngines);

    const engines = parseResult(result);

    yield put({ type: SET_ENGINES, payload: engines });

    yield put({ type: SET_ENGINES_SELECTED, payload: [] });
  } catch (error) {
    yield put({ type: FETCH_ENGINES_ERROR, error });
  }
}

export function* deleteEngine(id) {
  try {
    yield call(clientApi().deleteEngine, id);
  } catch (error) {
    const name = yield select(path(['entities', 'engines', 'byId', id, 'name']));
    return { isSuccess: false, name, id };
  }

  return { isSuccess: true, id };
}

export function* deleteEngines() {
  yield put({ type: PENDING_REQUEST });

  try {
    const selected = yield select(path(['navigation', 'engines', 'selected']));

    if (selected) {
      const deletedEnginesState = yield all(map(id => call(deleteEngine, id), selected));
      const errors = pipe(filter(propEq('isSuccess', false)), map(prop('name')))(deletedEnginesState);

      const hasError = !isNil(errors) && !isEmpty(errors);
      const hasMoreThanOneError = hasError && length(errors) > 1;
      if (hasError)
        yield put({
          type: DELETE_ENGINES_ERROR,
          error: {
            errorMessage: `Echec de suppression d${hasMoreThanOneError ? 'es' : 'u'}  moteur${
              hasMoreThanOneError ? 's' : ''
            } ${join(',', errors)}`,
          },
        });

      const success = pipe(filter(propEq('isSuccess', true)), map(prop('id')))(deletedEnginesState);

      if (isNil(success) || isEmpty(success)) {
        yield put({ type: RESET_PENDING_REQUEST });
        return null;
      }

      const engines = yield select(pipe(path(['entities', 'engines', 'byId']), values));
      const newEngines = !isNil(engines) && engines.filter(engine => !contains(prop('id', engine), success));
      yield put({ type: SET_ENGINES, payload: newEngines });

      yield put({ type: SET_ENGINES_SELECTED, payload: [] });
      if (!hasError) yield put({ type: DELETE_ENGINES_SUCCESS });
    }
  } catch (error) {
    yield put({
      type: DELETE_ENGINES_ERROR,
      error: { errorMessage: prop('message', error) },
    });
  }

  yield put({ type: RESET_PENDING_REQUEST });
  return null;
}

export default [takeLatest(FETCH_ENGINES, fetchEngines), takeLatest(DELETE_ENGINES_REQUEST, deleteEngines)];
