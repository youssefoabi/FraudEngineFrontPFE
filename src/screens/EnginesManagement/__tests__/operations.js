import { itReturnsAllMutations, itShouldMatchSequence, NO_PARAM } from '@cdiscount/jest-utils';

import { deleteEngine, deleteEngines, fetchEngines, parseResult } from '../operations';
import engines from '../__fixtures__';

jest.mock('react-global-configuration');

describe('EnginesManagement operations', () => {
  describe('parseResult', () => {
    itReturnsAllMutations(parseResult, [
      { name: 'without result', props: [null] },
      { name: 'with wrong result', props: [{ foo: {} }] },
      { name: 'with good result', props: [{ data: engines }] },
    ]);
  });
  describe('saga', () => {
    itShouldMatchSequence('fetchEngines', () => fetchEngines(), [NO_PARAM, { data: engines }]);
    itShouldMatchSequence('deleteEngines with success', () => deleteEngines(), [
      NO_PARAM,
      [1, 2],
      [true],
      [true],
    ]);
    itShouldMatchSequence('deleteEngines with failure', () => deleteEngines(), [
      NO_PARAM,
      [1, 2],
      [true],
      [false],
    ]);
    itShouldMatchSequence('deleteEngine with success', () => deleteEngine(1), [NO_PARAM]);
    itShouldMatchSequence('deleteEngine with failure', () => deleteEngine(1), [NO_PARAM, new Error()]);
  });
});
