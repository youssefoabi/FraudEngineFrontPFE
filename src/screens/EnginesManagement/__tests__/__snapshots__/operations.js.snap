// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnginesManagement operations parseResult with good result returns correct result 1`] = `
Array [
  Object {
    "id": 10,
    "isEnable": true,
    "modifiedAt": 2010-01-30T01:11:00.000Z,
    "name": "Engine 01",
    "scopes": Array [
      "gamme00",
    ],
    "urlTemplate": "stringA01",
  },
  Object {
    "id": 11,
    "isEnable": false,
    "modifiedAt": 2010-01-30T01:11:00.000Z,
    "name": "Engine 11",
    "scopes": Array [
      "gamme10",
      "gamme11",
    ],
    "urlTemplate": "stringA11",
  },
  Object {
    "id": 12,
    "isEnable": false,
    "modifiedAt": 2011-01-30T01:11:00.000Z,
    "name": "Engine 21",
    "scopes": Array [
      "gamme20",
      "gamme21",
      "gamme22",
    ],
    "urlTemplate": "stringA21",
  },
  Object {
    "id": 13,
    "isEnable": true,
    "modifiedAt": 2011-03-02T01:11:00.000Z,
    "name": "Engine 31",
    "scopes": Array [
      "gamme30",
      "gamme31",
      "gamme32",
      "gamme33",
    ],
    "urlTemplate": "stringA31",
  },
  Object {
    "id": 14,
    "isEnable": false,
    "modifiedAt": 2012-03-01T01:11:00.000Z,
    "name": "Engine 41",
    "scopes": Array [
      "gamme40",
      "gamme41",
      "gamme42",
      "gamme43",
      "gamme44",
    ],
    "urlTemplate": "stringA41",
  },
  Object {
    "id": 15,
    "isEnable": false,
    "modifiedAt": 2012-03-01T01:11:00.000Z,
    "name": "Engine 51",
    "scopes": Array [
      "gamme50",
      "gamme51",
      "gamme52",
      "gamme53",
      "gamme54",
      "gamme55",
    ],
    "urlTemplate": "stringA51",
  },
  Object {
    "id": 16,
    "isEnable": true,
    "modifiedAt": 2013-03-30T01:11:00.000Z,
    "name": "Engine 61",
    "scopes": Array [
      "gamme60",
      "gamme61",
      "gamme62",
      "gamme63",
      "gamme64",
      "gamme65",
      "gamme66",
    ],
    "urlTemplate": "stringA61",
  },
  Object {
    "id": 17,
    "isEnable": false,
    "modifiedAt": 2013-03-30T01:11:00.000Z,
    "name": "Engine 71",
    "scopes": Array [
      "gamme70",
      "gamme71",
      "gamme72",
      "gamme73",
      "gamme74",
      "gamme75",
      "gamme76",
      "gamme77",
    ],
    "urlTemplate": "stringA71",
  },
  Object {
    "id": 18,
    "isEnable": false,
    "modifiedAt": 2014-03-30T01:11:00.000Z,
    "name": "Engine 81",
    "scopes": Array [
      "gamme80",
      "gamme81",
      "gamme82",
      "gamme83",
      "gamme84",
      "gamme85",
      "gamme86",
      "gamme87",
      "gamme88",
    ],
    "urlTemplate": "stringA81",
  },
  Object {
    "id": 19,
    "isEnable": true,
    "modifiedAt": 2014-04-30T01:11:00.000Z,
    "name": "Engine 91",
    "scopes": Array [
      "gamme90",
      "gamme91",
      "gamme92",
      "gamme93",
      "gamme94",
      "gamme95",
      "gamme96",
      "gamme97",
      "gamme98",
      "gamme99",
    ],
    "urlTemplate": "stringA91",
  },
  Object {
    "id": 20,
    "isEnable": false,
    "modifiedAt": 2015-04-30T01:11:00.000Z,
    "name": "Engine 101",
    "scopes": Array [
      "gamme100",
      "gamme101",
      "gamme102",
      "gamme103",
      "gamme104",
      "gamme105",
      "gamme106",
      "gamme107",
      "gamme108",
      "gamme109",
      "gamme1010",
    ],
    "urlTemplate": "stringA101",
  },
  Object {
    "id": 21,
    "isEnable": false,
    "modifiedAt": 2015-04-30T01:11:00.000Z,
    "name": "Engine 111",
    "scopes": Array [
      "gamme110",
      "gamme111",
      "gamme112",
      "gamme113",
      "gamme114",
      "gamme115",
      "gamme116",
      "gamme117",
      "gamme118",
      "gamme119",
      "gamme1110",
      "gamme1111",
    ],
    "urlTemplate": "stringA111",
  },
  Object {
    "id": 22,
    "isEnable": true,
    "modifiedAt": 2016-05-30T01:11:00.000Z,
    "name": "Engine 121",
    "scopes": Array [
      "gamme120",
      "gamme121",
      "gamme122",
      "gamme123",
      "gamme124",
      "gamme125",
      "gamme126",
      "gamme127",
      "gamme128",
      "gamme129",
      "gamme1210",
      "gamme1211",
      "gamme1212",
    ],
    "urlTemplate": "stringA121",
  },
  Object {
    "id": 23,
    "isEnable": false,
    "modifiedAt": 2016-05-30T01:11:00.000Z,
    "name": "Engine 131",
    "scopes": Array [
      "gamme130",
      "gamme131",
      "gamme132",
      "gamme133",
      "gamme134",
      "gamme135",
      "gamme136",
      "gamme137",
      "gamme138",
      "gamme139",
      "gamme1310",
      "gamme1311",
      "gamme1312",
      "gamme1313",
    ],
    "urlTemplate": "stringA131",
  },
  Object {
    "id": 24,
    "isEnable": false,
    "modifiedAt": 2017-05-30T01:11:00.000Z,
    "name": "Engine 141",
    "scopes": Array [
      "gamme140",
      "gamme141",
      "gamme142",
      "gamme143",
      "gamme144",
      "gamme145",
      "gamme146",
      "gamme147",
      "gamme148",
      "gamme149",
      "gamme1410",
      "gamme1411",
      "gamme1412",
      "gamme1413",
      "gamme1414",
    ],
    "urlTemplate": "stringA141",
  },
  Object {
    "id": 25,
    "isEnable": true,
    "modifiedAt": 2017-06-30T01:11:00.000Z,
    "name": "Engine 151",
    "scopes": Array [
      "gamme150",
      "gamme151",
      "gamme152",
      "gamme153",
      "gamme154",
      "gamme155",
      "gamme156",
      "gamme157",
      "gamme158",
      "gamme159",
      "gamme1510",
      "gamme1511",
      "gamme1512",
      "gamme1513",
      "gamme1514",
      "gamme1515",
    ],
    "urlTemplate": "stringA151",
  },
  Object {
    "id": 26,
    "isEnable": false,
    "modifiedAt": 2018-06-30T01:11:00.000Z,
    "name": "Engine 161",
    "scopes": Array [
      "gamme160",
      "gamme161",
      "gamme162",
      "gamme163",
      "gamme164",
      "gamme165",
      "gamme166",
      "gamme167",
      "gamme168",
      "gamme169",
      "gamme1610",
      "gamme1611",
      "gamme1612",
      "gamme1613",
      "gamme1614",
      "gamme1615",
      "gamme1616",
    ],
    "urlTemplate": "stringA161",
  },
  Object {
    "id": 27,
    "isEnable": false,
    "modifiedAt": 2018-06-30T01:11:00.000Z,
    "name": "Engine 171",
    "scopes": Array [
      "gamme170",
      "gamme171",
      "gamme172",
      "gamme173",
      "gamme174",
      "gamme175",
      "gamme176",
      "gamme177",
      "gamme178",
      "gamme179",
      "gamme1710",
      "gamme1711",
      "gamme1712",
      "gamme1713",
      "gamme1714",
      "gamme1715",
      "gamme1716",
      "gamme1717",
    ],
    "urlTemplate": "stringA171",
  },
  Object {
    "id": 28,
    "isEnable": true,
    "modifiedAt": 2019-07-30T01:11:00.000Z,
    "name": "Engine 181",
    "scopes": Array [
      "gamme180",
      "gamme181",
      "gamme182",
      "gamme183",
      "gamme184",
      "gamme185",
      "gamme186",
      "gamme187",
      "gamme188",
      "gamme189",
      "gamme1810",
      "gamme1811",
      "gamme1812",
      "gamme1813",
      "gamme1814",
      "gamme1815",
      "gamme1816",
      "gamme1817",
      "gamme1818",
    ],
    "urlTemplate": "stringA181",
  },
]
`;

exports[`EnginesManagement operations parseResult with wrong result returns correct result 1`] = `Array []`;

exports[`EnginesManagement operations parseResult without result returns correct result 1`] = `Array []`;

exports[`EnginesManagement operations saga deleteEngine with failure 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          1,
        ],
        "context": null,
        "fn": [Function],
      },
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "SELECT": Object {
        "args": Array [],
        "selector": [Function],
      },
    },
  },
  Object {
    "done": true,
    "value": Object {
      "id": 1,
      "isSuccess": false,
      "name": null,
    },
  },
]
`;

exports[`EnginesManagement operations saga deleteEngine with success 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          1,
        ],
        "context": null,
        "fn": [Function],
      },
    },
  },
  Object {
    "done": true,
    "value": Object {
      "id": 1,
      "isSuccess": true,
    },
  },
]
`;

exports[`EnginesManagement operations saga deleteEngines with failure 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "type": "PENDING_REQUEST",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "SELECT": Object {
        "args": Array [],
        "selector": [Function],
      },
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "ALL": Array [
        Object {
          "@@redux-saga/IO": true,
          "CALL": Object {
            "args": Array [
              true,
            ],
            "context": null,
            "fn": [Function],
          },
        },
      ],
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "type": "RESET_PENDING_REQUEST",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": true,
    "value": null,
  },
]
`;

exports[`EnginesManagement operations saga deleteEngines with success 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "type": "PENDING_REQUEST",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "SELECT": Object {
        "args": Array [],
        "selector": [Function],
      },
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "ALL": Array [
        Object {
          "@@redux-saga/IO": true,
          "CALL": Object {
            "args": Array [
              true,
            ],
            "context": null,
            "fn": [Function],
          },
        },
      ],
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "type": "RESET_PENDING_REQUEST",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": true,
    "value": null,
  },
]
`;

exports[`EnginesManagement operations saga fetchEngines 1`] = `
Array [
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [],
        "context": null,
        "fn": [Function],
      },
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "payload": Array [
            Object {
              "id": 10,
              "isEnable": true,
              "modifiedAt": 2010-01-30T01:11:00.000Z,
              "name": "Engine 01",
              "scopes": Array [
                "gamme00",
              ],
              "urlTemplate": "stringA01",
            },
            Object {
              "id": 11,
              "isEnable": false,
              "modifiedAt": 2010-01-30T01:11:00.000Z,
              "name": "Engine 11",
              "scopes": Array [
                "gamme10",
                "gamme11",
              ],
              "urlTemplate": "stringA11",
            },
            Object {
              "id": 12,
              "isEnable": false,
              "modifiedAt": 2011-01-30T01:11:00.000Z,
              "name": "Engine 21",
              "scopes": Array [
                "gamme20",
                "gamme21",
                "gamme22",
              ],
              "urlTemplate": "stringA21",
            },
            Object {
              "id": 13,
              "isEnable": true,
              "modifiedAt": 2011-03-02T01:11:00.000Z,
              "name": "Engine 31",
              "scopes": Array [
                "gamme30",
                "gamme31",
                "gamme32",
                "gamme33",
              ],
              "urlTemplate": "stringA31",
            },
            Object {
              "id": 14,
              "isEnable": false,
              "modifiedAt": 2012-03-01T01:11:00.000Z,
              "name": "Engine 41",
              "scopes": Array [
                "gamme40",
                "gamme41",
                "gamme42",
                "gamme43",
                "gamme44",
              ],
              "urlTemplate": "stringA41",
            },
            Object {
              "id": 15,
              "isEnable": false,
              "modifiedAt": 2012-03-01T01:11:00.000Z,
              "name": "Engine 51",
              "scopes": Array [
                "gamme50",
                "gamme51",
                "gamme52",
                "gamme53",
                "gamme54",
                "gamme55",
              ],
              "urlTemplate": "stringA51",
            },
            Object {
              "id": 16,
              "isEnable": true,
              "modifiedAt": 2013-03-30T01:11:00.000Z,
              "name": "Engine 61",
              "scopes": Array [
                "gamme60",
                "gamme61",
                "gamme62",
                "gamme63",
                "gamme64",
                "gamme65",
                "gamme66",
              ],
              "urlTemplate": "stringA61",
            },
            Object {
              "id": 17,
              "isEnable": false,
              "modifiedAt": 2013-03-30T01:11:00.000Z,
              "name": "Engine 71",
              "scopes": Array [
                "gamme70",
                "gamme71",
                "gamme72",
                "gamme73",
                "gamme74",
                "gamme75",
                "gamme76",
                "gamme77",
              ],
              "urlTemplate": "stringA71",
            },
            Object {
              "id": 18,
              "isEnable": false,
              "modifiedAt": 2014-03-30T01:11:00.000Z,
              "name": "Engine 81",
              "scopes": Array [
                "gamme80",
                "gamme81",
                "gamme82",
                "gamme83",
                "gamme84",
                "gamme85",
                "gamme86",
                "gamme87",
                "gamme88",
              ],
              "urlTemplate": "stringA81",
            },
            Object {
              "id": 19,
              "isEnable": true,
              "modifiedAt": 2014-04-30T01:11:00.000Z,
              "name": "Engine 91",
              "scopes": Array [
                "gamme90",
                "gamme91",
                "gamme92",
                "gamme93",
                "gamme94",
                "gamme95",
                "gamme96",
                "gamme97",
                "gamme98",
                "gamme99",
              ],
              "urlTemplate": "stringA91",
            },
            Object {
              "id": 20,
              "isEnable": false,
              "modifiedAt": 2015-04-30T01:11:00.000Z,
              "name": "Engine 101",
              "scopes": Array [
                "gamme100",
                "gamme101",
                "gamme102",
                "gamme103",
                "gamme104",
                "gamme105",
                "gamme106",
                "gamme107",
                "gamme108",
                "gamme109",
                "gamme1010",
              ],
              "urlTemplate": "stringA101",
            },
            Object {
              "id": 21,
              "isEnable": false,
              "modifiedAt": 2015-04-30T01:11:00.000Z,
              "name": "Engine 111",
              "scopes": Array [
                "gamme110",
                "gamme111",
                "gamme112",
                "gamme113",
                "gamme114",
                "gamme115",
                "gamme116",
                "gamme117",
                "gamme118",
                "gamme119",
                "gamme1110",
                "gamme1111",
              ],
              "urlTemplate": "stringA111",
            },
            Object {
              "id": 22,
              "isEnable": true,
              "modifiedAt": 2016-05-30T01:11:00.000Z,
              "name": "Engine 121",
              "scopes": Array [
                "gamme120",
                "gamme121",
                "gamme122",
                "gamme123",
                "gamme124",
                "gamme125",
                "gamme126",
                "gamme127",
                "gamme128",
                "gamme129",
                "gamme1210",
                "gamme1211",
                "gamme1212",
              ],
              "urlTemplate": "stringA121",
            },
            Object {
              "id": 23,
              "isEnable": false,
              "modifiedAt": 2016-05-30T01:11:00.000Z,
              "name": "Engine 131",
              "scopes": Array [
                "gamme130",
                "gamme131",
                "gamme132",
                "gamme133",
                "gamme134",
                "gamme135",
                "gamme136",
                "gamme137",
                "gamme138",
                "gamme139",
                "gamme1310",
                "gamme1311",
                "gamme1312",
                "gamme1313",
              ],
              "urlTemplate": "stringA131",
            },
            Object {
              "id": 24,
              "isEnable": false,
              "modifiedAt": 2017-05-30T01:11:00.000Z,
              "name": "Engine 141",
              "scopes": Array [
                "gamme140",
                "gamme141",
                "gamme142",
                "gamme143",
                "gamme144",
                "gamme145",
                "gamme146",
                "gamme147",
                "gamme148",
                "gamme149",
                "gamme1410",
                "gamme1411",
                "gamme1412",
                "gamme1413",
                "gamme1414",
              ],
              "urlTemplate": "stringA141",
            },
            Object {
              "id": 25,
              "isEnable": true,
              "modifiedAt": 2017-06-30T01:11:00.000Z,
              "name": "Engine 151",
              "scopes": Array [
                "gamme150",
                "gamme151",
                "gamme152",
                "gamme153",
                "gamme154",
                "gamme155",
                "gamme156",
                "gamme157",
                "gamme158",
                "gamme159",
                "gamme1510",
                "gamme1511",
                "gamme1512",
                "gamme1513",
                "gamme1514",
                "gamme1515",
              ],
              "urlTemplate": "stringA151",
            },
            Object {
              "id": 26,
              "isEnable": false,
              "modifiedAt": 2018-06-30T01:11:00.000Z,
              "name": "Engine 161",
              "scopes": Array [
                "gamme160",
                "gamme161",
                "gamme162",
                "gamme163",
                "gamme164",
                "gamme165",
                "gamme166",
                "gamme167",
                "gamme168",
                "gamme169",
                "gamme1610",
                "gamme1611",
                "gamme1612",
                "gamme1613",
                "gamme1614",
                "gamme1615",
                "gamme1616",
              ],
              "urlTemplate": "stringA161",
            },
            Object {
              "id": 27,
              "isEnable": false,
              "modifiedAt": 2018-06-30T01:11:00.000Z,
              "name": "Engine 171",
              "scopes": Array [
                "gamme170",
                "gamme171",
                "gamme172",
                "gamme173",
                "gamme174",
                "gamme175",
                "gamme176",
                "gamme177",
                "gamme178",
                "gamme179",
                "gamme1710",
                "gamme1711",
                "gamme1712",
                "gamme1713",
                "gamme1714",
                "gamme1715",
                "gamme1716",
                "gamme1717",
              ],
              "urlTemplate": "stringA171",
            },
            Object {
              "id": 28,
              "isEnable": true,
              "modifiedAt": 2019-07-30T01:11:00.000Z,
              "name": "Engine 181",
              "scopes": Array [
                "gamme180",
                "gamme181",
                "gamme182",
                "gamme183",
                "gamme184",
                "gamme185",
                "gamme186",
                "gamme187",
                "gamme188",
                "gamme189",
                "gamme1810",
                "gamme1811",
                "gamme1812",
                "gamme1813",
                "gamme1814",
                "gamme1815",
                "gamme1816",
                "gamme1817",
                "gamme1818",
              ],
              "urlTemplate": "stringA181",
            },
          ],
          "type": "@ENGINES/SET",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": false,
    "value": Object {
      "@@redux-saga/IO": true,
      "PUT": Object {
        "action": Object {
          "payload": Array [],
          "type": "@ENGINES/SET_SELECTED",
        },
        "channel": null,
      },
    },
  },
  Object {
    "done": true,
    "value": undefined,
  },
]
`;
